set default scaling equal across all noise plots

bw1129 opened the pull request:
I would really like to see default scaling set equal across all noise plots. Sometimes people upload these plots and are mislead by the fact that the post filtered gyro looks noisier than prefiltered, because the scaling is not comparable across plots. I realize the colorbar is there to help, but for practical purposes, it's rather misleading at a glance, and it is currently not possible to really make a direct clear comparison even using the colorbar. By scaling them the same, the differences in density of yellow hot spots is a true representation of the differences. Even if this results are over saturated for the pre-filtered gyro (debug) condition, this cond mainly serves as a benchmark anyway. 

It only requires a small change to line 398: lims=np.array([[1, 20],[1, 20], [1, 20], [0,meanspec_max*1.5]])
I think 1 to 20 is perfect across all. I think line 398 contains the default lims. Looks like it is set to detect the max value it sees for each axis and scale accordingly. Maybe there's a better way to just create another option called "equal-scaling" and make that default, and leave the current "auto" option if someone wants to input that.

bw1129 committed:
set default scaling equal across all noise plots

set default scaling equal across all noise plots.

bw1129 committed:
Update PID-Analyzer.py

bw1129 reviewed (commented):
Im proposing a couple of small changes here, the main one being a change in the default scaling (line 394) making the z-axis the same across all panels of the noise spectrograms.

Bradw-17 commented:
This is a more acceptable scale fix? If so I will delete my PR and add these changes to my own repo.

bw1129 commented:
> This is a more acceptable scale fix? If so I will delete my PR and add these changes to my own repo.

This proposal is just to fix the scales the same across all panels. You can still input any scale you want in the command window if that's what you want, but I thought you needed a change specific to Unix or Mac environment? In any case, I hope they consider it, cause it would certainly clear up a lot of confusion some people have.

bw1129 committed:
display warning for incorrect debug_mode

I'd like to propose that the debug noise plots display a warning if the debug variable does not contain prefiltered gyro data. This would help avoid the confusion some people have when they have unknowingly selected the incorrect debug_mode in Betaflight, resulting in meaningless results in the debug subplots. An easy way to do this is to code when the 4th column of debug[3] is not empty indicating debug variable does not contain prefiltered gyro data. We could consider using debug_mode variable itself, but this wont work for certain earlier versions of BF. The solution I propose here involves 2 simple lines of code, the first (line 703) checks to see if debug[3] contains data (indicating incorrect debug_mode, since 4th column should be empty otherwise), and the second is a conditional statement (line 470) for presenting warning overlaid on debug subplots. There might be a more elegant way to do this, but this works.

bw1129 committed:
Update PID-Analyzer.py
